cellxpredict/session.py

```diff
@@ -0,0 +1,73 @@
+import json
+
+from pathlib import Path
+from typing import Union
+
+from .config import ConfigBase
+
+
+def write_config_json_file(config: ConfigBase) -> None:
+    """Record params of the training run."""
+    # extract the params into dict:
+    json_data = {prm : str(getattr(config, prm)) for prm in config.__dict__}
+
+    # write the data into json file:
+    # model = str(config.model).capitalize()
+    # json_fn = config.model_dir / f'ConfigHyperParams-{model}.json'
+    json_fn = config.model_dir / 'ConfigHyperParams.json'
+    with open(json_fn, 'w') as json_file:
+        json.dump(json_data, json_file, indent=4)
+
+
+def load_config_attributes(
+    default_config: ConfigBase,
+    json_params_file: Union[str, Path]
+) -> ConfigBase:
+    """Overwrite default config params from JSON file.
```

Comment:
I let this function to work for any config inheriting from the `ConfigBase` object, so this can work for the downstream models and not just the encoder. 

**Challenge**: currently, this fn cannot be defined in `cellx` because it requires a cellx-predict specific object, `ConfigBase`. 

**Future**: Perhaps we could make this pipeline to work in a step-wise manner rather than to train all parts in one go (like the full model). Eventually, I guess it would be reasonable to, say, train a projector on top of a previously trained encoder etc.